src/Core/Metadata/AttributeTrait.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Metadata;
+
+/**
+ * @experimental
+ */
+trait AttributeTrait
```

Comment:
Do we really need to keep this magic? As it's mostly of internal use, using the `extraProperties` is probably good enough.